docs/aws/audit/lambdamonitoring/rules/calls_in_cloudtrail.mdx

```diff
@@ -24,6 +24,228 @@ AWSWAF, PCIDSS
+
+   ```
+   aws lambda list-functions --region your-region-name
+   ```
+   Replace 'your-region-name' with the name of your AWS region. This command will return a list of all the Lambda functions in the specified region.
+
+3. Check CloudTrail for each Lambda function: For each Lambda function, you need to check if CloudTrail is enabled. You can do this by using the following AWS CLI command:
+
+   ```
+   aws cloudtrail describe-trails
+   ```
+   This command will return a list of all the CloudTrail trails in your AWS account. You need to check if the Lambda function name appears in the list of trails.
+
+4. Check CloudTrail event selectors: For each trail, you need to check if the event selectors are configured to include Lambda events. You can do this by using the following AWS CLI command:
+
+   ```
+   aws cloudtrail get-event-selectors --trail-name your-trail-name
+   ```
+   Replace 'your-trail-name' with the name of your CloudTrail trail. This command will return the event selectors for the specified trail. You need to check if the 'IncludeManagementEvents' is set to true and 'ReadWriteType' is set to 'All'. If these conditions are not met, then CloudTrail is not properly configured for the Lambda function.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary AWS SDK for Python (Boto3) modules in your Python script. You will need the 'boto3' module to interact with AWS services and the 'botocore.exceptions' module to handle any exceptions that may occur.
+
+```python
+import boto3
+from botocore.exceptions import NoCredentialsError, ClientError
+```
+
+2. Create a session using your AWS credentials. If you have your AWS credentials set up in your environment, Boto3 will automatically use those. If not, you can manually input your access key, secret access key, and region.
+
+```python
+try:
+    session = boto3.Session(
+        aws_access_key_id='YOUR_ACCESS_KEY',
+        aws_secret_access_key='YOUR_SECRET_KEY',
+        region_name='YOUR_REGION'
+    )
+except NoCredentialsError:
+    print("No AWS credentials were found.")
+```
+
+3. Create a client for AWS Lambda and AWS CloudTrail. You will use the Lambda client to list all your Lambda functions and the CloudTrail client to check if CloudTrail is enabled for each function.
+
+```python
+lambda_client = session.client('lambda')
+cloudtrail_client = session.client('cloudtrail')
+```
+
+4. Iterate over all your Lambda functions and check if CloudTrail is enabled for each one. If CloudTrail is not enabled for a function, print a message indicating this.
+
+```python
+try:
+    functions = lambda_client.list_functions()
+    for function in functions['Functions']:
+        try:
+            trails = cloudtrail_client.describe_trails(trailNameList=[function['FunctionName']])
+            if not trails['trailList']:
+                print(f"CloudTrail is not enabled for the function: {function['FunctionName']}")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>